Smith.MatrixSdk.Example/Application.cs

```diff
@@ -1,6 +1,7 @@
-﻿using System;
```

Comment:
It appears that all source files have a BOM marker. Is it a .NET convention? Should I preserve it?

Reply:
Today, that's more of my oversight than a convention.

See: earlier days, all the .NET tooling was following the Windows sort-of-convention that says that files that have a BOM (either UTF-8 or a UTF-16 one) are Unicode-encoded, and all the other ones are considered ANSI-encoded by default (where "ANSI" is actually some random system-local Windows code page, like CP1252 on English systems). So, if the systems of your developers working on a project are configured to use different "ANSI" code pages, then the developers will get different compilation results, which is never a good thing.

On the other hand, UTF-8 is really the way to go, so most of the editors suited for .NET development historically have been configured to either place a UTF-8 BOM on every file they create, or at least offer this option when noticing any characters outside of a 7-bit ANSI range in a user-edited file.

[Rider](https://www.jetbrains.com/rider/), which I am a user and a developer of, has fallen a victim of this plague, too: by default, it is configured to add a UTF-8 BOM to every new file when running on Windows, for better compatibility with older tooling (considering there're no older, "ANSI"-oriented tooling on other OS).

These days, it doesn't make too much sense for people working on modern versions of .NET, since it follows the convention of accepting UTF-8 by default. So, the BOMs aren't required, and I'm going to remove them all from the repo using a script (and will also add a shared Rider settings file to override the default and not add a BOM to any new files – which will only affect people working on the project using Rider, of course). You're free to omit them from any new files.

Thanks for noticing!